routing.go

```diff
@@ -177,7 +200,11 @@ func (dht *IpfsDHT) GetValues(ctx context.Context, key string, nvals int) (_ []R
 
 	// If we have it local, don't bother doing an RPC!
 	lrec, err := dht.getLocal(key)
-	if err == nil {
+	if err != nil {
```

Comment:
Can this if statement include the logic from the one on line 219?

Reply:
No... that logic is wrong.